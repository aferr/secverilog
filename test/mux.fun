; Domain(0)=D1, Domain(1)=D2
(declare-fun Domain (Int) Label)
(assert (= (Domain 0) D1))
(assert (= (Domain 1) D2))
