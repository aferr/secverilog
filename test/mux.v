module Two_Input_Mux
(
	input {D1}          in1,
	input {D2}          in2,
	input {L}           sel,
	output {Domain sel} out
);

reg	{Domain sel} out;

always @ (*) begin
	if (sel == 1'b0)
		out = in1;
	else
		out = in2;
end
endmodule
