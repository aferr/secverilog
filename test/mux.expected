Compiling file mux.v
Verifying file mux.v
verified
Total: 0 assertions failed
